src/plot_api/plot_api.js

```diff
@@ -399,11 +400,6 @@ function setPlotContext(gd, config) {
             }
         });
 
-        // cause a remake of the modebar any time we change context
-        if(gd._fullLayout && gd._fullLayout._modebar) {
-            delete gd._fullLayout._modebar;
```

Comment:
no need for this catch anymore. `manageModebar` and `Modebar.update` take care of this.